src/main/java/net/eriagacha/CustomItem.java

```diff
@@ -0,0 +1,43 @@
+package net.eriagacha;
+
+import lombok.extern.log4j.Log4j2;
+import net.eriagacha.network.NetworkClient;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Hand;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.world.World;
+
+@Log4j2
+public class CustomItem extends Item {
+
+  public CustomItem(Settings settings) {
+    super(settings);
+  }
+
+  @Override
+  public TypedActionResult<ItemStack> use(World world, PlayerEntity user, Hand hand) {
+    if (this.isFood()) {
+      ItemStack itemStack = user.getStackInHand(hand);
+      if (user.canConsume(this.getFoodComponent().isAlwaysEdible())) {
+        user.setCurrentHand(hand);
+        if (FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT) {
```

Comment:
A helper with this check as isClient or isServer would be good.

Reply:
Done boy